plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -356,6 +374,41 @@ public boolean isInUse(InetAddress addr, int port) {
     }
   }
 
+  @VisibleForTesting
+  static void putAppEngineApiSdkJarIntoApps(List<File> appDirectories,
+      ILibraryRepositoryService repositoryService) throws CoreException {
+    try {
+      Library library = CloudLibraries.getLibrary("appengine-api");
+      for (LibraryFile libraryFile : library.getAllDependencies()) {
+        Artifact artifact =
+            repositoryService.resolveArtifact(libraryFile, new NullProgressMonitor());
+        Path artifactPath = artifact.getFile().toPath();
+
+        for (File appDirectory : appDirectories) {
+          Path libDirectory = appDirectory.toPath().resolve("WEB-INF/lib");
+          libDirectory.toFile().mkdirs();
+          if (!appEngineApiSdkJarExists(libDirectory)) {
+            Files.copy(artifactPath, libDirectory.resolve(artifactPath.getFileName()));
+          }
+        }
+      }
+    } catch (IOException e) {
+      String message = Messages.getString("cannot.copy.appengine.api.sdk.jar");
+      throw new CoreException(StatusUtil.error(LocalAppEngineServerBehaviour.class, message, e));
+    }
+  }
+
+  @VisibleForTesting
+  static boolean appEngineApiSdkJarExists(Path directory) {
+    for (String file : directory.toFile().list()) {
+      String lowercase = file.toLowerCase(Locale.US);
```

Comment:
should work in practice. Could use a regexp instead, but I'm not sure it matters